### PR TITLE
Mjulian31

### PR DESCRIPTION
**Pull request by mjulian31**
@vchuravy, here are the tests and implementation for CUDA @ka_code_typed support

### CODE REVIEW

File: src/reflection.jl [hunk 1]
```diff
@@ -17,25 +23,58 @@ function ka_code_typed(kernel, argtypes; ndrange=nothing, workgroupsize=nothing,
 end
 
 
+"""
+Get the llvm code for a kernel
```
Comment (vchuravy) [inline src/reflection.jl:27]: This is the typed IR and not LLVM IR yet.

File: src/reflection.jl [hunk 2]
```diff
@@ -17,25 +23,58 @@ function ka_code_typed(kernel, argtypes; ndrange=nothing, workgroupsize=nothing,
...
+
+# Examples
+```
+@ka_code_typed kernel(args)
```
Comment (vchuravy) [inline src/reflection.jl:31]:
```suggestion
@ka_code_typed kernel(args. ndrange=...)
```

File: src/reflection.jl [hunk 3]
```diff
@@ -17,25 +23,58 @@ function ka_code_typed(kernel, argtypes; ndrange=nothing, workgroupsize=nothing,
...
+    @assert(old_args != nothing)
+    @assert(kern != nothing)
 
     thecall = InteractiveUtils.gen_call_with_extracted_types_and_kwargs(__module__, :ka_code_typed, ex)
 
     quote
+        local $(esc(args)) = $(old_args)
+        if isa($kern, Kernel{CUDADevice})
+            # translate CuArray to CuDeviceArray
+            local $(esc(args)) = map(CUDA.cudaconvert, $(esc(args)))
```
Comment (vchuravy) [inline src/reflection.jl:75]:
```suggestion
            $(esc(args)) = map(CUDA.cudaconvert, $(esc(args)))
```
I don't think you need the `local` here

**Review by vchuravy — COMMENTED** [review]
The git history is a bit messy, but I can squash that on merge.

**Comment by jakebolewski** [comment]
Bump.  It would be nice to get this in as it solves most of the issues I was having with trying this out on the ClimateMachine.

**Comment by mjulian31** [comment]
Changes pushed, all tests still passing.

**Comment by vchuravy** [comment]
Thanks!

**Comment by jakebolewski** [comment]
Thanks @mjulian31! This is a great contribution that helps a lot with debugging.

**Pull request merged**